snapcraft/internal/project_loader/grammar_processing/_part_grammar_processor.py

```diff
@@ -58,6 +58,12 @@ def __init__(self, *, plugin, properties, project_options, repo):
         self._project_options = project_options
         self._repo = repo
 
+        for key, value in properties.items():
```

Comment:
Actually modifying the properties here instead of providing accessor methods to get the processed grammar doesn't fit into the pattern established in this class and may have further ramifications. Can you explain why you chose this path?

Reply:
I was under the impression we'd discussed this and you said wasn't necessary to wrap the properties, but I think I got something wrong. So I went ahead and added an explicit get_properties() method now. Sorry for the mix-up.